api/mraa/aio.hpp

```diff
@@ -47,14 +47,27 @@ class Aio
      *
      * @param pin channel number to read ADC inputs
      */
-    Aio(unsigned int pin)
+    Aio(int pin)
```

Comment:
I dont' want this split out because otherwise the main commit won't work. Having it on it's own doesn't make sense imho because the reason for it is the main commit. Kind of a catch 22.

@Hbrinj I'd recommend you split it like this:
1. mraa_io_init C only
2. C++ template interface
3. Swig stuff on top